### PR TITLE
Read secrets from Amazon Secrets Manager

### PR DESCRIPTION
**Pull request by spilchen**
This is a refactor of the cloud package. I am taking out the MultiSourceSecretFetcher from the package and moving it to a new package. The new package will have less dependencies, like vapi, ctrl, and logr. The goal is that this new package will be included other products like vcluster and the nma. This will cut down on code duplication, as the similar logic is needed in these other products.

I also added support to read secrets from AWS Secrets Manager. The secret name must have a special path reference followed by the full ARN of the secret in the following formate: awssm://<secret-arn>. This can be used for the following secrets: communal.credentialSecret, nmaTLSCert and passwordSecret.

### CODE REVIEW

File: pkg/cloud/secret.go [hunk 1]
```diff
@@ -17,34 +17,31 @@ package cloud
...
+type ControllerSecretFetcher struct {
 	client.Client
 	Log      logr.Logger
 	EVWriter events.EVWriter
 	VDB      *vapi.VerticaDB
 }
 
 // Fetch reads the secret from a secret store. The contents of the secret is successful.
-func (m *MultiSourceSecretFetcher) Fetch(ctx context.Context, secretName types.NamespacedName) (map[string][]byte, error) {
+func (m *ControllerSecretFetcher) Fetch(ctx context.Context, secretName types.NamespacedName) (map[string][]byte, error) {
```
Comment (roypaulin) [inline pkg/cloud/secret.go:44]: minor
```suggestion
func (c *ControllerSecretFetcher) Fetch(ctx context.Context, secretName types.NamespacedName) (map[string][]byte, error) {
```

File: pkg/cloud/secret.go [hunk 2]
```diff
@@ -17,34 +17,31 @@ package cloud
...
 )
 
-// MultiSourceSecretFetcher is secret reader that handles retrival from
-// different sources such as Kubernetes secret store and Google Secrets Manager
-// (GSM).
-type MultiSourceSecretFetcher struct {
+// ControllerSecretFetcher is secret reader designed for the verticadb
+// controller. It can handle retrival from different sources, such as Kubernetes
+// secret store, Google Secrets Manager (GSM), etc.
+type ControllerSecretFetcher struct {
```
Comment (roypaulin) [inline pkg/cloud/secret.go:36]: Isn't `VerticaDBSecretFetcher` more accurate?

**Review by roypaulin — COMMENTED** [review]
Looks good!

File: pkg/cloud/secret.go (hunk 2 above)
Comment (spilchen) [reply pkg/cloud/secret.go]: I like this name.

File: pkg/secrets/fetcher.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+// out the different secret stores.
+type NotFoundError struct {
+	msg string
+}
+
+func (e *NotFoundError) Error() string {
+	return e.msg
+}
+
+// Fetch reads the secret from a secret store. The contents of the secret is successful.
```
[the remark this answers is not in the document]
Comment (spilchen) [reply pkg/secrets/fetcher.go]: I missed a word. I updated it to be: The contents of the secret is returned on success.

**Review by roypaulin — APPROVED** [review]

**Pull request merged**